tasks/eva_2399/mongo_count_validation.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+from argparse import ArgumentParser
+from datetime import datetime
+
+import psycopg2
+import psycopg2.extras
+from ebi_eva_common_pyutils.config_utils import get_pg_metadata_uri_for_eva_profile
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.mongodb import MongoDatabase
+from ebi_eva_common_pyutils.pg_utils import execute_query
+from retry import retry
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+mongo_migration_count_validation_table_name = "eva_tasks.mongo4_migration_count_validation"
+
+
+def create_collection_count_validation_report(mongo_source: MongoDatabase, mongo_dest: MongoDatabase, database_list):
+    count_validation_res_list = []
+
+    for db in database_list:
+        mongo_source.db_name = db
+        mongo_dest.db_name = db
+        source_collections = mongo_source.get_collection_names()
+        dest_collections = mongo_dest.get_collection_names()
+
+        if not source_collections and not dest_collections:
+            logger.info(f"database {db} does not exit in either of mongo instances")
+            continue
+
+        all_collections = sorted(set(source_collections).union(set(dest_collections)))
+
+        for coll in all_collections:
+            logger.info(f"fetching count for database {db} - collection {coll}")
+
+            no_of_documents_in_src = get_documents_count_for_collection(mongo_source, db, coll)
+            no_of_documents_in_dest = get_documents_count_for_collection(mongo_dest, db, coll)
+
+            if (no_of_documents_in_src == 0) and (coll not in source_collections):
+                logger.info(f"collection {coll} does not exist in database {db} in mongo source")
+                no_of_documents_in_src = -1
+
+            if (no_of_documents_in_dest == 0) and (coll not in dest_collections):
+                logger.info(f"collection {coll} does not exist in database {db} in dest source")
+                no_of_documents_in_dest = -1
+
+            if no_of_documents_in_src != no_of_documents_in_dest and \
+                    no_of_documents_in_src != -1 and no_of_documents_in_dest != -1:
+                logger.info(f"no of documents in collection {coll} of database {db} does not match")
```

Comment:
Could you add the actual number to the log statement?

Reply:
added